### PR TITLE
Write the ReadOnlySequence as a single ConnectionDataMessage

### LINKED ISSUE #124: "Message is incomplete" error
Hello! I just tried turning on the Azure SignalR Service and am running into an issue with large messages being sent from the server to the client. I am getting the following JS error message:

```
signalr.js:1616 Uncaught Error: Message is incomplete.
    at Function.TextMessageFormat.parse (signalr.js:1616)
    at JsonHubProtocol.parseMessages (signalr.js:3207)
    at HubConnection.processIncomingData (signalr.js:2158)
    at WebSocketTransport.HubConnection.connection.onreceive (signalr.js:1934)
    at WebSocket.webSocket.onmessage (signalr.js:2761)
```

I added a console.log to that TextMessageFormat.parse method to see the input, and I see that it is a C# object that is being broken into half. That is, the first message is of length 2,049 characters. Then that exception is thrown. Then I immediately get the next message that is about the same length (2,046 characters), followed by the same exception.

My method of passing the data to the front-end, from the hub, follows:

```
await Clients.Client(Context.ConnectionId).SendAsync("FullRsvpListOfAttendees", attendees.Select(n => new
{
   AttendeeId = n.AttendeeId,
   FirstName = n.AttendeeFirstName,
   LastName = n.AttendeeLastName,
   IsMuted = n.IsMuted
}).ToList());
```

I have attached some logs generated from the client and server, if that is useful.

When I run this same code using the base AspNetCore.SignalR, it works without this error message.

Here's a longer copy/paste from the JS console:
```
Debug: Server handshake complete.
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 202.
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 41.
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 44.
signalr.js:1893 Trace: (WebSockets transport) sending data. String data of length 76.
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 186.
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 44.
signalr.js:1893 Trace: (WebSockets transport) sending data. String data of length 89.
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 2048.
signalr.js:1616 Uncaught Error: Message is incomplete.
    at Function.TextMessageFormat.parse (signalr.js:1616)
    at JsonHubProtocol.parseMessages (signalr.js:3207)
    at HubConnection.processIncomingData (signalr.js:2158)
    at WebSocketTransport.HubConnection.connection.onreceive (signalr.js:1934)
    at WebSocket.webSocket.onmessage (signalr.js:2761)
TextMessageFormat.parse @ signalr.js:1616
JsonHubProtocol.parseMessages @ signalr.js:3207
HubConnection.processIncomingData @ signalr.js:2158
HubConnection.connection.onreceive @ signalr.js:1934
webSocket.onmessage @ signalr.js:2761
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 2046.
signalr.js:1616 Uncaught Error: Message is incomplete.
    at Function.TextMessageFormat.parse (signalr.js:1616)
    at JsonHubProtocol.parseMessages (signalr.js:3207)
    at HubConnection.processIncomingData (signalr.js:2158)
    at WebSocketTransport.HubConnection.connection.onreceive (signalr.js:1934)
    at WebSocket.webSocket.onmessage (signalr.js:2761)
TextMessageFormat.parse @ signalr.js:1616
JsonHubProtocol.parseMessages @ signalr.js:3207
HubConnection.processIncomingData @ signalr.js:2158
HubConnection.connection.onreceive @ signalr.js:1934
webSocket.onmessage @ signalr.js:2761
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 2048.
signalr.js:1616 Uncaught Error: Message is incomplete.
    at Function.TextMessageFormat.parse (signalr.js:1616)
    at JsonHubProtocol.parseMessages (signalr.js:3207)
    at HubConnection.processIncomingData (signalr.js:2158)
    at WebSocketTransport.HubConnection.connection.onreceive (signalr.js:1934)
    at WebSocket.webSocket.onmessage (signalr.js:2761)
TextMessageFormat.parse @ signalr.js:1616
JsonHubProtocol.parseMessages @ signalr.js:3207
HubConnection.processIncomingData @ signalr.js:2158
HubConnection.connection.onreceive @ signalr.js:1934
webSocket.onmessage @ signalr.js:2761
signalr.js:1893 Trace: (WebSockets transport) data received. String data of length 1176.
VM210868:1 Uncaught SyntaxError: Unexpected token , in JSON at position 0
    at JSON.parse (<anonymous>)
    at JsonHubProtocol.parseMessages (signalr.js:3211)
    at HubConnection.processIncomingData (signalr.js:2158)
    at WebSocketTransport.HubConnection.connection.onreceive (signalr.js:1934)
    at WebSocket.webSocket.onmessage (signalr.js:2761)
```

Thank you!

[Server Log.txt](https://github.com/Azure/azure-signalr/files/2103727/Server.Log.txt)
[Client Log.txt](https://github.com/Azure/azure-signalr/files/2103728/Client.Log.txt)
[localhost.har.txt](https://github.com/Azure/azure-signalr/files/2103729/localhost.har.txt)

### PR DESCRIPTION
**Pull request by davidfowl**
- Manually write the message pack header and ReadOnlySequence to the Stream directly.

Fixes #124 

PS: I need to write tests. We also need to update the logic in the service that writes to the transport.

### CODE REVIEW

File: src/Microsoft.Azure.SignalR/HubHost/ServiceConnection.cs [hunk 1]
```diff
@@ -460,20 +460,7 @@ private Task DispatchMessageAsync(ServiceMessage message)
...
-                            {
-                                // This is a multi-segmented buffer so just write each chunk
-                                // TODO: Optimize this by doing it all under a single lock
-                                var position = buffer.Start;
-                                while (buffer.TryGet(ref position, out var memory))
-                                {
-                                    await WriteAsync(new ConnectionDataMessage(connection.ConnectionId, memory));
-                                }
-                            }
+                            await WriteAsync(new ConnectionDataMessage(connection.ConnectionId, buffer));
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR/HubHost/ServiceConnection.cs]: This is the main change. We write the (potentially multi segment) message directly to the connection output.

File: src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs [hunk 2]
```diff
@@ -83,13 +84,18 @@ public class ConnectionDataMessage : ConnectionMessage
         /// <param name="connectionId">The connection Id.</param>
         /// <param name="payload">Binary data to be delivered.</param>
         public ConnectionDataMessage(string connectionId, ReadOnlyMemory<byte> payload) : base(connectionId)
+        {
+            Payload = new ReadOnlySequence<byte>(payload);
+        }
+
```
Comment (xscript) [inline src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: nit: API document.

File: src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs [hunk 3]
```diff
@@ -83,13 +84,18 @@ public class ConnectionDataMessage : ConnectionMessage
...
+        public ConnectionDataMessage(string connectionId, ReadOnlySequence<byte> payload) : base(connectionId)
         {
             Payload = payload;
         }
 
         /// <summary>
         /// Gets or sets the binary payload.
         /// </summary>
-        public ReadOnlyMemory<byte> Payload { get; set; }
+        public ReadOnlySequence<byte> Payload { get; set; }
```
Comment (xscript) [inline src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: This is a breaking change, right?

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: Yes. it is.

File: src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs (hunk 2 above)
Comment (davidfowl) [reply src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: Done

**Pull request merged**